### PR TITLE
Port hashlib changes from ipy2

### PR DESCRIPTION
**Pull request by slozier**

### CODE REVIEW

File: Src/IronPython.Modules/hashlib/_sha256.cs [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+
+            [Documentation("copy() -> object (copy of this object)")]
+            public override HashBase<SHA256> copy() {
+                Sha256Object res = new Sha256Object();
+                res._hasher = CloneHasher();
+                return res;
+            }
+
+            protected override void CreateHasher() {
+#if SILVERLIGHT || WP75
```
Comment (gpetrou) [inline Src/IronPython.Modules/hashlib/_sha256.cs]: Can we remove these?

Comment (slide) [inline Src/IronPython.Modules/hashlib/_sha256.cs]: Yes, but let's make it a different PR
